src/models/util/ParseCalChart3File.ts

```diff
@@ -0,0 +1,251 @@
+/**
+ * CalChart3 was a bytestream with a series of 4 char codes describing the next section.
+ * In general, the layout is <4char> <size_of_data> <data...> <"END "> <4char>, with
+ * the exception of the first "INGL" block and the version
+ * the Version block is either "GURK", for CalChart3.0, or "GUXY" where X and Y are version 
+ * numbers in hex.
+ * 
+ * For CalChart 3.3 and later there are consistent 4char blocks that contain more elements
+ * of the show.
+ */
+
+import Show from '../Show';
+import StuntSheet from '../StuntSheet';
+import StuntSheetDot from '../StuntSheetDot';
+
+/**
+ * LoadShow
+ * 
+ * @param buffer The Byte array of data to load.
+ * @returns Returns either a new [Show] or null if the Byte array is ill-formed.
+ */
+export default function LoadShow(buffer : ArrayBuffer) {
+  try {
+    // parsing frequently involves throwing to indicate the parse failed.
+    return LoadShowHelper(buffer);
+  }
+  catch(e) {
+    console.log(e);
+  }
+  return null;
+}
+
+function LoadShowHelper(buffer : ArrayBuffer) {
+  // first we parse and confirm the file is well formed
+  if (!ReadAndCheckID(new DataView(buffer, 0, 4), 0, "INGL")) {
+    return null;
```

Comment:
I think this may be ok.  This is saying that the file type isn't a *.shw file, not that the *.shw file is corrupted.  So I think null is the right thing to do here.

I throw later when it's rather challenging to have null propagate up the stack.

Reply:
I'm going to switch to throwing errors so they make their way up the stack to the User and we display them in the dialog.  So no more returning null.